test/lib/samson/secrets/vault_server_test.rb

```
@@ -49,6 +49,15 @@
         to_raise: Vault::HTTPError.new("address", stub(code: '200'))
       ) { refute_valid server }
     end
+
+    it "nullifies the association on destroy" do
+      deploy_group = deploy_groups(:pod100)
+      server.deploy_groups = [deploy_group]
+      server.save
```

`save!` so we don't get false-positives if the `save` silently failed